podman: broken on x86_64-darwin

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

x86_64-darwin build fails (aarch64-darwin or any linux don't).

### Steps to reproduce

```
$ git log --oneline | head -n 1
ac60d61e3737 pmbootstrap: 3.3.2 -> 3.4.0 (#407265)

$ nix build --file . podman --argstr system x86_64-darwin
error: builder for '/nix/store/baiblxdqk1yaxsmrds48mqa598p99n87-podman-5.4.1.drv' failed with exit code 2;
       last 25 log lines:
       > ./contrib/cirrus/mac_setup.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash"
       > ./contrib/cirrus/runner.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash"
       > ./contrib/cirrus/mac_env.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash"
       > ./contrib/cirrus/logcollector.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash"
       > ./contrib/cirrus/pr-should-link-jira: interpreter directive changed from "#!/bin/bash" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash"
       > ./contrib/cirrus/postbuild.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash"
       > ./contrib/cirrus/setup_environment.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash"
       > ./contrib/containers-common/containers-common.sh: interpreter directive changed from "#!/usr/bin/bash -e" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash -e"
       > ./.github/actions/check_cirrus_cron/cron_failures.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash"
       > ./.github/actions/check_cirrus_cron/rerun_failed_tasks.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash"
       > ./.github/actions/check_cirrus_cron/make_email_body.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash"
       > ./pkg/machine/compression/testdata/gen-testdata.sh: interpreter directive changed from "#!/bin/bash" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash"
       > ./pkg/machine/e2e/scripts/fake_compose: interpreter directive changed from "#!/bin/bash" to "/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash"
       > substituteStream() in derivation podman-5.4.1: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'Makefile')
       > substituteStream() in derivation podman-5.4.1: WARNING: pattern /bin/bash doesn't match anything in file 'Makefile'
       > mkdir -p bin/darwin
       > /nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash: line 1: git: command not found
       > CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 go build \
       >   \
       >      -ldflags ' -X github.com/containers/podman/v5/libpod/define.buildInfo=315532800  -X github.com/containers/podman/v5/libpod/config._installPrefix=/nix/store/74afy3ck9j9gvlwb83z0nkwnk43vb416-podman-5.4.1 -X github.com/containers/podman/v5/libpod/config._etcDir=/etc -X github.com/containers/podman/v5/pkg/systemd/quadlet._binDir=/nix/store/74afy3ck9j9gvlwb83z0nkwnk43vb416-podman-5.4.1/bin -X github.com/containers/common/pkg/config.additionalHelperBinariesDir=/nix/store/74afy3ck9j9gvlwb83z0nkwnk43vb416-podman-5.4.1/libexec/podman ' \
       >  -tags "remote exclude_graphdriver_btrfs btrfs_noversion exclude_graphdriver_devicemapper containers_image_openpgp" \
       >    -o bin/darwin/podman ./cmd/podman
       > # github.com/containers/podman/v5/cmd/podman
       > link: duplicated definition of symbol dlopen, from github.com/ebitengine/purego and github.com/ebitengine/purego
       > make: *** [Makefile:394: bin/darwin/podman] Error 1
       For full logs, run:
         nix log /nix/store/baiblxdqk1yaxsmrds48mqa598p99n87-podman-5.4.1.drv
```

### Expected behaviour

build works

### Screenshots

_No response_

### Relevant log output

```console

```

### Additional context

_No response_

### System metadata

```
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.4.0, macOS 15.4.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.28.3`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/yhc8a0a2mvbp8fp53l57i3d5cnz735fc-source`
```

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/po/podman/package.nix
+++ b/pkgs/by-name/po/podman/package.nix
@@ -74,13 +74,13 @@ let
 in
 buildGoModule rec {
   pname = "podman";
-  version = "5.4.1";
+  version = "5.5.1";
 
   src = fetchFromGitHub {
     owner = "containers";
     repo = "podman";
     rev = "v${version}";
-    hash = "sha256-RirMBb45KeBLdBJrRa86WxI4FiXdBar+RnVQ2ezEEYc=";
+    hash = "sha256-/dGFDwjAAc1D88VslVDolf2YVPZ9cHUCQjdaEreQSE0=";
   };
 
   patches = [

